task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,281 @@
+        optional: true
+  steps:
+    - name: sast-unicode-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.6@sha256:5f298d8d990dfa82023e50029b71b08e19c3c9cedb181dfc4bc86c9ecad8700c
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+      env:
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: FIND_UNICODE_CONTROL_GIT_URL
+          value: $(params.FIND_UNICODE_CONTROL_GIT_URL)
+        - name: FIND_UNICODE_CONTROL_ARGS
+          value: $(params.FIND_UNICODE_CONTROL_ARGS)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: SOURCE_CODE_DIR
+          value: $(workspaces.workspace.path)
+      script: |
+        #!/usr/bin/env bash
+        set -exuo pipefail
+        . /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        SCAN_PROP=""
+
+        ca_bundle=/mnt/trusted-ca/ca-bundle.crt
+        if [ -f "$ca_bundle" ]; then
+          echo "INFO: Using mounted CA bundle: $ca_bundle"
+          cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+          update-ca-trust
+        fi
+
+        # Clone the source code from upstream repo
+        clone_unicontrol()
+        {
+            GIT_URL=$(echo "${FIND_UNICODE_CONTROL_GIT_URL}" | awk -F'#' '{print $1}')
+            REV=$(echo "${FIND_UNICODE_CONTROL_GIT_URL}" | awk -F'#' '{print $2}')
+
+            # Clone find-unicode-control repository
+            CLONE_EXIT_CODE=0
+            git clone "${GIT_URL}" find-unicode-control >&2 || CLONE_EXIT_CODE=$?
+            if [[ "${CLONE_EXIT_CODE}" -eq 0 ]]; then
+                echo "Message: Succeed to clone the repository: ${GIT_URL}" >&2
+            else
+                echo "Error: Failed to clone the repository: ${GIT_URL}" >&2
+                return 1
+            fi
+
+            if [[ -n "${REV}" ]]; then
+                # Get git url suffix
+                git_url_suffix=$(git -C ./find-unicode-control/ rev-parse HEAD)
+                if [[ "${git_url_suffix}" != "${REV}" ]]; then
+                    git -C ./find-unicode-control/ checkout --quiet "${REV}"
+                fi
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}"
+            else
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}#${git_url_suffix}"
+            fi
+
+            # Check usage of find-unicode-control to confirm the tool can work
+            USAGE_EXIT_CODE=0
+            ./find-unicode-control/find_unicode_control.py --help >&2 || USAGE_EXIT_CODE=$?
+            if [[ "${USAGE_EXIT_CODE}" -eq 0 ]]; then
+                echo "Message: Succeed to check usage of find-unicode-control" >&2
```

Comment:
I would drop **all** the code that prints `Message: Succeed to ...`.  The whole script is executed with `set -x`, so it is clear what is being executed and we expect all these commands to succeed under normal circumstances.  So it is sufficient just to handle the failures, in my opinion.

Reply:
I also find the handling of failures inconsistent.  The exit code of `git checkout --quiet ...` is not explicitly checked despite its success highly depends on the value of the provided `KFP_GIT_URL`.  In contrast to this, the failure of `sed -i` command is very unlikely and its explicit handling just complicates the shell script.